gravitee-node-api/src/main/java/io/gravitee/node/api/cache/Cache.java

```diff
@@ -32,107 +32,317 @@
+     * @param key the key.
+     * @param value the value.
+     *
+     * @return a <code>Maybe</code> containing the previous value or an empty <code>Maybe</code> if no value was present.
+     */
+    default Maybe<V> rxPut(final K key, final V value) {
+        return Maybe.fromCallable(() -> this.put(key, value)).subscribeOn(Schedulers.io());
     }
 
+    /**
+     * Same as {@link #put(Object, Object)} but with a time to live.
+     * @param key the key.
+     * @param value the value.
+     *
+     * @return the previous value associated to the specified key or <code>null</code> if no value was present.
+     */
     V put(final K key, final V value, final long ttl, final TimeUnit ttlUnit);
 
-    default Single<V> rxPut(final K key, final V value, final long ttl, final TimeUnit ttlUnit) {
-        return Single.fromCallable(() -> this.put(key, value, ttl, ttlUnit)).subscribeOn(Schedulers.io());
+    /**
+     * Same as {@link #rxPut(Object, Object)} but with a time to live.
+     * @param key the key.
+     * @param value the value.
+     *
+     * @return the previous value associated to the specified key or <code>null</code> if no value was present.
+     */
+    default Maybe<V> rxPut(final K key, final V value, final long ttl, final TimeUnit ttlUnit) {
+        return Maybe.fromCallable(() -> this.put(key, value, ttl, ttlUnit)).subscribeOn(Schedulers.io());
     }
 
+    /**
+     * Put all the specified key/value pairs into this cache.
+     * @param m a map containing all the key/value pairs to put into this cache.
+     */
     void putAll(final Map<? extends K, ? extends V> m);
 
+    /**
+     * Reactive method to put all the specified key/value pairs into this cache.
+     * @param m a map containing all the key/value pairs to put into this cache.
+     *
+     * @return a <code>Completable</code> that completes once all the key/value pairs have been put in the cache.
+     */
     default Completable rxPutAll(final Map<? extends K, ? extends V> m) {
         return Completable.fromRunnable(() -> this.putAll(m)).subscribeOn(Schedulers.io());
     }
 
-    V computeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction);
-
-    default Single<V> rxComputeIfAbsent(K key, Function<? super K, ? extends V> mappingFunction) {
-        return Single.fromCallable(() -> this.computeIfAbsent(key, mappingFunction)).subscribeOn(Schedulers.io());
+    /**
+     * If the specified key is not already associated with a value, attempt to compute the value of the specified key using the given remapping function.
+     * If the remapping function returns <code>null</code>, the mapping is removed.
+     * @param key the key.
+     * @param remappingFunction the function that is executed to associate the new value in case no value was present.
+     *
+     * @return the new value if no value was present or the existing one if it was already present.
```

Comment:
It should return the new mapping value (including null) in any case.

Reply:
As seen together, the expected behavior is the following

`@return the computed value if it was absent (<code>null</code> if the mapping function returns <code>null</code>) or the existing one if it was already present.`